sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusReceiver.cs

```diff
@@ -225,6 +226,33 @@ public class ServiceBusReceiver : IAsyncDisposable
             return messages;
         }
 
+        /// <summary>
+        /// Receives messages as an asynchronous enumerable from the entity using <see cref="ReceiveMode"/> mode.
+        /// <see cref="ReceiveMode"/> defaults to PeekLock mode. Messages will be received from the entity as
+        /// the IAsyncEnumerable is iterated. If no messages are available, this method will continue polling
+        /// until messages are available, i.e. it will never return null.
+        /// </summary>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the
+        /// request to cancel the operation.</param>
+        /// <returns>The message received.</returns>
+        public virtual async IAsyncEnumerable<ServiceBusReceivedMessage> ReceiveMessagesAsync(
+            [EnumeratorCancellation]
+            CancellationToken cancellationToken = default)
```

Comment:
If `cancellationToken` is `default`, the iterator below will run forever. Should we check allow `default` value here?

Reply:
Yes; it's a valid scenario for a processing application to run forever until the host process is torn down.  So long as messages are flowing, callers may `break` if ending the loop is desired.